src/mono/mono/metadata/assembly.c

```diff
@@ -82,8 +83,9 @@ mono_assemblies_unlock (void)
 
 /* If defined, points to the bundled assembly information */
 static const MonoBundledAssembly **bundles;
-
+static gboolean hash_contains_bundled_assemblies;
```

Comment:
It could be in `bundled-resources.c`. The main point was to have a parallel to `bundles` and `satellite_bundles` that could be leveraged in the search paths. I figured it would be slightly faster to just return a static bool from `assembly.c` than to query `bundled-resources.c` for every bundle search api, and it was a parallel to how `bundles` is set in `mono_register_bundled_assemblies` and how `satellite_bundles` is set in `mono_register_bundled_satellite_assemblies`. Would it be preferred to have a getter function that returns a bool value over returning the bool value directly?

Reply:
> Would it be preferred to have a getter function that returns a bool value over returning the bool value directly?

I don't particularly care about that.

My main concerns are:

1. I don't want to see logically separate runtime subsystems entangled (assembly definition, assembly probing on the one hand, vs bundle searching on the other)
2. Setting a global static in assembly.c from within bundled-resources.c - for a value that notionally doesn't ever change - either bundles are there or not.  feels wrong.  But I care less about this, to be honest.

Primiarly I would like the mechanics of how bundles work hidden away.  assembly.c is already pretty massive and complicated.  I would like to see code move out of it, if possible, not into it.

Reply:
For example: I'd love for `open_from_bundle_internal` (and associated code like `bundled_assembly_match`) to be moved out of assembly.c

that it is there right now is technical debt. not a design choice.